internal/js/modules/k6/browser/common/element_handle.go

```diff
@@ -1673,24 +1690,38 @@ func (h *ElementHandle) newPointerAction(
 func retryPointerAction(
 	apiCtx context.Context, fn retryablePointerActionFunc, opts *ElementHandleBasePointerOptions,
 ) (res any, err error) {
-	// try the default scrolling
-	if res, err = fn(apiCtx, nil); opts.Force || err == nil {
-		return res, err
-	}
-	// try with different scrolling options
-	for _, p := range []ScrollPosition{
-		ScrollPositionStart,
-		ScrollPositionCenter,
-		ScrollPositionEnd,
-		ScrollPositionNearest,
-	} {
-		s := ScrollIntoViewOptions{Block: p, Inline: p}
-		if res, err = fn(apiCtx, &s); err == nil {
-			break
+	for {
+		res, err = fn(apiCtx, nil)
+		if opts.Force || err == nil {
+			return res, err
 		}
-	}
 
-	return res, err
+		// try with different scrolling options
+		for _, p := range []ScrollPosition{
+			ScrollPositionStart,
+			ScrollPositionCenter,
+			ScrollPositionEnd,
+			ScrollPositionNearest,
+		} {
+			s := ScrollIntoViewOptions{Block: p, Inline: p}
+			if res, err = fn(apiCtx, &s); err == nil {
+				return res, nil
+			}
+		}
+
+		if !strings.Contains(err.Error(), "check if element is visible") &&
+			!strings.Contains(err.Error(), "error:notvisible") {
+			return res, err
+		}
+
+		// Wait with timeout or context cancellation
+		select {
+		case <-apiCtx.Done():
+			return nil, apiCtx.Err()
+		case <-time.After(20 * time.Millisecond):
+			// Continue retrying after delay
+		}
```

Comment:
Is `opts.Timeout` somehow considered here (e.g. implicitly as part of `apiCtx.Done() handling)? Otherwise, I think we should handle it explicitly.

Reply:
Yeah, the context with a timeout is used here. The timeout context is setup [here](https://github.com/grafana/k6/blob/dfedf4beeed4d344306598c8374e4c0641270c12/internal/js/modules/k6/browser/common/helpers.go#L100-L103).